src/components/borrow/BorrowAnvil.tsx

```diff
@@ -22,49 +30,27 @@ export const BorrowAnvil: React.FC<ReserveComponentProps> = ({ poolId, assetId }
   const userPoolData = useStore((state) => state.userPoolData.get(poolId));
   const reserve = poolData?.reserves.get(assetId);
   const assetToBase = reserve?.oraclePrice ?? 1;
-  const baseToAsset = 1 / assetToBase;
 
   const [toBorrow, setToBorrow] = useState<string>('');
-
+  const [simResult, setSimResult] = useState<ContractResponse<Positions>>();
   const decimals = reserve?.config.decimals ?? 7;
   const symbol = reserve?.tokenMetadata?.symbol ?? '';
 
-  // calculate current wallet state
-  const curBorrowed = userPoolData?.estimates?.liabilities?.get(assetId) ?? 0;
-  const oldBorrowCap = userPoolData
-    ? userPoolData.estimates.totalEffectiveCollateral -
-      userPoolData.estimates.totalEffectiveLiabilities
-    : undefined;
-  const oldBorrowCapAsset =
-    reserve && oldBorrowCap ? oldBorrowCap * baseToAsset * reserve.getLiabilityFactor() : undefined;
-  const oldBorrowLimit = userPoolData
-    ? userPoolData.estimates.totalEffectiveLiabilities /
-      userPoolData.estimates.totalEffectiveCollateral
-    : undefined;
-
-  // calculate new wallet state
-  let num_borrow = 0;
-  let newEffectiveLiabilities = 0;
-  if (toBorrow && userPoolData && reserve) {
-    num_borrow = Number(toBorrow);
-    let borrow_base = num_borrow * assetToBase * reserve.getLiabilityFactor();
-    newEffectiveLiabilities = userPoolData.estimates.totalEffectiveLiabilities + borrow_base;
-  }
-  const borrowCap = userPoolData
-    ? userPoolData.estimates.totalEffectiveCollateral - newEffectiveLiabilities
-    : undefined;
-  const borrowCapAsset =
-    reserve && borrowCap ? borrowCap * baseToAsset * reserve.getLiabilityFactor() : undefined;
-  const borrowLimit = userPoolData
-    ? newEffectiveLiabilities / userPoolData.estimates.totalEffectiveCollateral
-    : undefined;
-  const newAssetUtil = reserve
-    ? (reserve.estimates.borrowed + num_borrow) / reserve.estimates.supplied
-    : 0;
-
   if (txStatus === TxStatus.SUCCESS && Number(toBorrow) != 0) {
     setToBorrow('0');
   }
+  useDebouncedState(toBorrow, 500, async () => {
+    let sim = await handleSubmitTransaction(true);
```

Comment:
if the api is not that fast then there could be an edge case in which the following happens

type -> 500ms pass and api gets called but no simResult yet -> you type again -> 500ms pass again -> firstSimResult happens -> second simResult happens 

should'nt be much of an issue unless we're expecting apis to be slow but i think its a "bug" we can live with 

the solution would be api call cancelling which is not really fun to handle 🤣